extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -0,0 +1,126 @@
+package io.quarkus.kubernetes.deployment;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import io.ap4k.Session;
+import io.ap4k.SessionWriter;
+import io.ap4k.kubernetes.annotation.KubernetesApplication;
+import io.ap4k.kubernetes.generator.DefaultKubernetesApplicationGenerator;
+import io.ap4k.processor.SimpleFileWriter;
+import io.ap4k.project.Project;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationInfoBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.kubernetes.spi.KubernetesPortBuildItem;
+
+class KubernetesProcessor {
+
+    @Inject
+    BuildProducer<GeneratedResourceBuildItem> generatedResourceProducer;
+
+    @Inject
+    BuildProducer<FeatureBuildItem> featureProducer;
+
+    @BuildStep
+    public void build(ApplicationInfoBuildItem applicationInfo,
+            KubernetesConfig kubernetesConfig,
+            List<KubernetesPortBuildItem> kubernetesPortBuildItems) {
+
+        if (kubernetesPortBuildItems.isEmpty()) {
+            return;
+        }
+
+        // The resources that ap4k's execution will result in, will later-on be written
+        // by quarkus in the 'wiring-classes' directory
+        // The location is needed in order to properly support s2i build triggering
+
+        // by passing false to SimpleFileWriter, we ensure that no files are actually written during this phase
+        final Path root;
+        try {
+            root = Files.createTempDirectory("quarkus-kubernetes");
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to setup environment for generating Kubernetes resources", e);
+        }
+
+        final SessionWriter sessionWriter = new SimpleFileWriter(root, false);
+        sessionWriter.setProject(new Project());
+        final Session session = Session.getSession();
+        session.setWriter(sessionWriter);
+
+        final Map<String, Integer> ports = verifyPorts(kubernetesPortBuildItems);
+        enableKubernetes(applicationInfo, kubernetesConfig, ports);
+
+        // write the generated resources to the filesystem
+        final Map<String, String> generatedResourcesMap = session.close();
+        for (String generatedResourceFullPath : generatedResourcesMap.keySet()) {
+            generatedResourceProducer.produce(
+                    new GeneratedResourceBuildItem(
+                            // we need to make sure we are only passing the relative path to the build item
+                            generatedResourceFullPath.replace(root.toAbsolutePath() + "/", "META-INF/ap4k/"),
```

Comment:
Since `ap4k` is an implementation detail, maybe rename this to just `META-INF/kubernetes`?

Reply:
Indeed, I will change it

Reply:
Fixed